### DISCUSSION TITLE
在setu条件里加上keyword后一直提示没有符合条件的色图

### DISCUSSION THREAD

**Discussion by Eudiplozoon** [Tsuk1ko/CQ-picfinder-robot#267] [问题求助]
一开始几张加了keyword的是可以发出来的，后面稍微调了一下config就报这个错了

logs如下

> 2021/12/4 下午11:22:04 [error] oy.mk shorten
      Error: Request failed with status code 400
    at createError (/root/pics/cq-picsearcher-bot/node_modules/axios/lib/core/createError.js:16:15)
    at settle (/root/pics/cq-picsearcher-bot/node_modules/axios/lib/core/settle.js:17:12)
    at IncomingMessage.handleStreamEnd (/root/pics/cq-picsearcher-bot/node_modules/axios/lib/adapters/http.js:293:11)
    at IncomingMessage.emit (node:events:406:35)
    at IncomingMessage.emit (node:domain:470:12)
    at endReadableNT (node:internal/streams/readable:1329:12)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  config: {
    transitional: {
      silentJSONParsing: true,
      forcedJSONParsing: true,
      clarifyTimeoutError: false
    },
    adapter: [Function: httpAdapter],
    transformRequest: [ [Function: transformRequest] ],
    transformResponse: [ [Function: transformResponse] ],
    timeout: 0,
    xsrfCookieName: 'XSRF-TOKEN',
    xsrfHeaderName: 'X-XSRF-TOKEN',
    maxContentLength: -1,
    maxBodyLength: -1,
    validateStatus: [Function: validateStatus],
    headers: {
      Accept: 'application/json, text/plain, */*',
      'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.79 Safari/537.36'
    },
    method: 'get',
    url: 'https://oy.mk/api/insert?url=https%3A%2F%2Fi.pixiv.cat%2Fimg-original%2Fimg%2F2018%2F01%2F29%2F14%2F24%2F42%2F67021147_p4.jpg',
    data: undefined
  },
  request: <ref *1> ClientRequest {
    _events: [Object: null prototype] {
      abort: [Function (anonymous)],
      aborted: [Function (anonymous)],
      connect: [Function (anonymous)],
      error: [Function (anonymous)],
      socket: [Function (anonymous)],
      timeout: [Function (anonymous)],
      prefinish: [Function: requestOnPrefinish]
    },
    _eventsCount: 7,
    _maxListeners: undefined,
    outputData: [],
    outputSize: 0,
    writable: true,
    destroyed: false,
    _last: true,
    chunkedEncoding: false,
    shouldKeepAlive: false,
    _defaultKeepAlive: true,
    useChunkedEncodingByDefault: false,
    sendDate: false,
    _removedConnection: false,
    _removedContLen: false,
    _removedTE: false,
    _contentLength: 0,
    _hasBody: true,
    _trailer: '',
    finished: true,
    _headerSent: true,
    _closed: false,
    socket: TLSSocket {
      _tlsOptions: [Object],
      _secureEstablished: true,
      _securePending: false,
      _newSessionPending: false,
      _controlReleased: true,
      secureConnecting: false,
      _SNICallback: null,
      servername: 'oy.mk',
      alpnProtocol: false,
      authorized: true,
      authorizationError: null,
      encrypted: true,
      _events: [Object: null prototype],
      _eventsCount: 10,
      connecting: false,
      _hadError: false,
      _parent: null,
      _host: 'oy.mk',
      _readableState: [ReadableState],
      _maxListeners: undefined,
      _writableState: [WritableState],
      allowHalfOpen: false,
      _sockname: null,
      _pendingData: null,
      _pendingEncoding: '',
      server: undefined,
      _server: null,
      ssl: [TLSWrap],
      _requestCert: true,
      _rejectUnauthorized: true,
      parser: null,
      _httpMessage: [Circular *1],
      [Symbol(res)]: [TLSWrap],
      [Symbol(verified)]: true,
      [Symbol(pendingSession)]: null,
      [Symbol(async_id_symbol)]: 316119,
      [Symbol(kHandle)]: [TLSWrap],
      [Symbol(kSetNoDelay)]: false,
      [Symbol(lastWriteQueueSize)]: 0,
      [Symbol(timeout)]: null,
      [Symbol(kBuffer)]: null,
      [Symbol(kBufferCb)]: null,
      [Symbol(kBufferGen)]: null,
      [Symbol(kCapture)]: false,
      [Symbol(kBytesRead)]: 0,
      [Symbol(kBytesWritten)]: 0,
      [Symbol(connect-options)]: [Object],
      [Symbol(RequestTimeout)]: undefined
    },
    _header: 'GET /api/insert?url=https%3A%2F%2Fi.pixiv.cat%2Fimg-original%2Fimg%2F2018%2F01%2F29%2F14%2F24%2F42%2F67021147_p4.jpg HTTP/1.1\r\n' +
      'Accept: application/json, text/plain, */*\r\n' +
      'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.79 Safari/537.36\r\n' +
      'Host: oy.mk\r\n' +
      'Connection: close\r\n' +
      '\r\n',
    _keepAliveTimeout: 0,
    _onPendingData: [Function: nop],
    agent: Agent {
      _events: [Object: null prototype],
      _eventsCount: 2,
      _maxListeners: undefined,
      defaultPort: 443,
      protocol: 'https:',
      options: [Object: null prototype],
      requests: [Object: null prototype] {},
      sockets: [Object: null prototype],
      freeSockets: [Object: null prototype] {},
      keepAliveMsecs: 1000,
      keepAlive: false,
      maxSockets: Infinity,
      maxFreeSockets: 256,
      scheduling: 'lifo',
      maxTotalSockets: Infinity,
      totalSocketCount: 1,
      maxCachedSessions: 100,
      _sessionCache: [Object],
      [Symbol(kCapture)]: false
    },
    socketPath: undefined,
    method: 'GET',
    maxHeaderSize: undefined,
    insecureHTTPParser: undefined,
    path: '/api/insert?url=https%3A%2F%2Fi.pixiv.cat%2Fimg-original%2Fimg%2F2018%2F01%2F29%2F14%2F24%2F42%2F67021147_p4.jpg',
    _ended: true,
    res: IncomingMessage {
      _readableState: [ReadableState],
      _events: [Object: null prototype],
      _eventsCount: 3,
      _maxListeners: undefined,
      socket: [TLSSocket],
      httpVersionMajor: 1,
      httpVersionMinor: 1,
      httpVersion: '1.1',
      complete: true,
      rawHeaders: [Array],
      rawTrailers: [],
      aborted: false,
      upgrade: false,
      url: '',
      method: null,
      statusCode: 400,
      statusMessage: 'Bad Request',
      client: [TLSSocket],
      _consuming: false,
      _dumped: false,
      req: [Circular *1],
      responseUrl: 'https://oy.mk/api/insert?url=https%3A%2F%2Fi.pixiv.cat%2Fimg-original%2Fimg%2F2018%2F01%2F29%2F14%2F24%2F42%2F67021147_p4.jpg',
      redirects: [],
      [Symbol(kCapture)]: false,
      [Symbol(kHeaders)]: [Object],
      [Symbol(kHeadersCount)]: 10,
      [Symbol(kTrailers)]: null,
      [Symbol(kTrailersCount)]: 0,
      [Symbol(RequestTimeout)]: undefined
    },
    aborted: false,
    timeoutCb: null,
    upgradeOrConnect: false,
    parser: null,
    maxHeadersCount: null,
    reusedSocket: false,
    host: 'oy.mk',
    protocol: 'https:',
    _redirectable: Writable {
      _writableState: [WritableState],
      _events: [Object: null prototype],
      _eventsCount: 2,
      _maxListeners: undefined,
      _options: [Object],
      _ended: true,
      _ending: true,
      _redirectCount: 0,
      _redirects: [],
      _requestBodyLength: 0,
      _requestBodyBuffers: [],
      _onNativeResponse: [Function (anonymous)],
      _currentRequest: [Circular *1],
      _currentUrl: 'https://oy.mk/api/insert?url=https%3A%2F%2Fi.pixiv.cat%2Fimg-original%2Fimg%2F2018%2F01%2F29%2F14%2F24%2F42%2F67021147_p4.jpg',
      [Symbol(kCapture)]: false
    },
    [Symbol(kCapture)]: false,
    [Symbol(kNeedDrain)]: false,
    [Symbol(corked)]: 0,
    [Symbol(kOutHeaders)]: [Object: null prototype] {
      accept: [Array],
      'user-agent': [Array],
      host: [Array]
    }
  },
  response: {
    status: 400,
    statusText: 'Bad Request',
    headers: {
      server: 'nginx/1.18.0',
      date: 'Sat, 04 Dec 2021 15:22:04 GMT',
      'content-type': 'application/json',
      'content-length': '61',
      connection: 'close'
    },
    config: {
      transitional: [Object],
      adapter: [Function: httpAdapter],
      transformRequest: [Array],
      transformResponse: [Array],
      timeout: 0,
      xsrfCookieName: 'XSRF-TOKEN',
      xsrfHeaderName: 'X-XSRF-TOKEN',
      maxContentLength: -1,
      maxBodyLength: -1,
      validateStatus: [Function: validateStatus],
      headers: [Object],
      method: 'get',
      url: 'https://oy.mk/api/insert?url=https%3A%2F%2Fi.pixiv.cat%2Fimg-original%2Fimg%2F2018%2F01%2F29%2F14%2F24%2F42%2F67021147_p4.jpg',
      data: undefined
    },
    request: <ref *1> ClientRequest {
      _events: [Object: null prototype],
      _eventsCount: 7,
      _maxListeners: undefined,
      outputData: [],
      outputSize: 0,
      writable: true,
      destroyed: false,
      _last: true,
      chunkedEncoding: false,
      shouldKeepAlive: false,
      _defaultKeepAlive: true,
      useChunkedEncodingByDefault: false,
      sendDate: false,
      _removedConnection: false,
      _removedContLen: false,
      _removedTE: false,
      _contentLength: 0,
      _hasBody: true,
      _trailer: '',
      finished: true,
      _headerSent: true,
      _closed: false,
      socket: [TLSSocket],
      _header: 'GET /api/insert?url=https%3A%2F%2Fi.pixiv.cat%2Fimg-original%2Fimg%2F2018%2F01%2F29%2F14%2F24%2F42%2F67021147_p4.jpg HTTP/1.1\r\n' +
        'Accept: application/json, text/plain, */*\r\n' +
        'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.79 Safari/537.36\r\n' +
        'Host: oy.mk\r\n' +
        'Connection: close\r\n' +
        '\r\n',
      _keepAliveTimeout: 0,
      _onPendingData: [Function: nop],
      agent: [Agent],
      socketPath: undefined,
      method: 'GET',
      maxHeaderSize: undefined,
      insecureHTTPParser: undefined,
      path: '/api/insert?url=https%3A%2F%2Fi.pixiv.cat%2Fimg-original%2Fimg%2F2018%2F01%2F29%2F14%2F24%2F42%2F67021147_p4.jpg',
      _ended: true,
      res: [IncomingMessage],
      aborted: false,
      timeoutCb: null,
      upgradeOrConnect: false,
      parser: null,
      maxHeadersCount: null,
      reusedSocket: false,
      host: 'oy.mk',
      protocol: 'https:',
      _redirectable: [Writable],
      [Symbol(kCapture)]: false,
      [Symbol(kNeedDrain)]: false,
      [Symbol(corked)]: 0,
      [Symbol(kOutHeaders)]: [Object: null prototype]
    },
    data: { code: 400, message: '该网站存在风险！', data: null }
  },
  isAxiosError: true,
  toJSON: [Function: toJSON]

  }

    

设置项
   

>  // setu 相关功能，部分配置的详细说明请查看“wiki-附加功能-setu”以了解更多
   "setu": {
      // 是否启用
      "enable": true,
      // APIKEY（已弃用，不再需要）
      "apikey": "",
      // 反和谐方式（0：关闭，1：轻微修改，2：旋转）
      "antiShielding": 1,
      // 是否允许私聊使用
      "allowPM": true,
      // P站图本地反代服务主机，可以手动指定，一般不需要动，详见 wiki
      "pximgServerHost": "127.0.0.1",
      // P站图本地反代服务端口，可以手动指定，0 为自动分配，详见 wiki
      "pximgServerPort": 0,
      // 发送时使用的P站本地反代服务地址，一般不需要，详见 wiki
      "usePximgAddr": "",
      // 发送时请求图片使用的P站在线反代服务地址，支持占位符 {{pid}} {{p}} {{uid}} {{ext}} {{path}}，详见 wiki（使用时上面三个配置将不再起效）
      "pximgProxy": "",
      // 镜像地址，发送图片时一起发送的在线反向代理的原图地址，设置为空数组则不追加，支持占位符同上
      "sendPximgProxys": [
        "https://i.pixiv.cat/"
      ],
      // 对镜像地址进行短链接处理，可选值: ["oy.mk", "yww.uy"]，留空 "" 则不处理。
      "shortenPximgProxy": "",
      // 是否发送 master1200 大小的图片，以节省流量或加快发送速度，关闭时将发送原图
      "size1200": false,
      // 发送后这么多秒自动撤回（0 则不撤回，-1 为发送闪照）
      "deleteTime": 120,
      // 群内使用冷却时间（秒），每名用户独立，0 则无冷却，私聊无 cd
      "cd": 20,
      // 每名用户每日次数限制
      "limit": 50,
      // 群组白名单
      "whiteGroup": [ ],
      // 仅允许白名单群使用（与上面的私聊使用是独立的）
      "whiteOnly": false,
      // 白名单群组的群内使用冷却时间（秒），每名用户独立，0 则无冷却
      "whiteCd": 10,
      // 白名单群组内发送后这么多秒自动撤回（0 则不撤回，-1 为发送闪照）
      "whiteDeleteTime": 120,
      // r18 是否只发送地址不发送图片（如果开启，建议设置 `sendPximgProxys`）
      "r18OnlyUrl": {
        "private": false, // 私聊
        "group": false, // 群聊
        "temp": false // 群临时会话
      },
      // r18 图片只通过私聊发送（私聊不会被和谐，因此不会进行反和谐）
      "r18OnlyPrivate": false,
      // 允许临时会话发送 r18 结果，用法同 `bot.pmSearchResultTemp`
      "r18OnlyPrivateAllowTemp": false,
      // 只允许在白名单群组中发送 r18 setu
      "r18OnlyInWhite": true
    },`

**Comment by Tsuk1ko** [comment]
看起来你设置了 `shortenPximgProxy`，oy.mk 报 400 了，不要设置就好了，记得重载配置

**Reply by Eudiplozoon** [reply]
我把shortenPximgProxy删除并且重载和npm restart之后还是同样报错，似乎那项留空之后还是会根据之前设置的来处理短链

**Reply by Eudiplozoon** [reply]
重新部署后用原来的config文件试了一下还是同样情况，但是不会留下logs了
